hypervisor/hw/pci.c

```diff
@@ -207,20 +211,26 @@ static uint32_t pci_mmcfg_read_cfg(union pci_bdf bdf, uint32_t offset, uint32_t
  */
 static void pci_mmcfg_write_cfg(union pci_bdf bdf, uint32_t offset, uint32_t bytes, uint32_t val)
 {
-	uint32_t addr = mmcfg_off_to_address(bdf, offset);
-	void *hva = hpa2hva(addr);
+	uint32_t addr;
+	void *hva;
 
-	ASSERT(pci_is_valid_access(offset, bytes), "the offset should be aligned with 2/4 byte\n");
-	switch (bytes) {
-	case 1U:
-		mmio_write8((uint8_t)val, hva);
-		break;
-	case 2U:
-		mmio_write16((uint16_t)val, hva);
-		break;
-	default:
-		mmio_write32(val, hva);
-		break;
+	if (pci_is_valid_access(offset, bytes)) {
+		addr = mmcfg_off_to_address(bdf, offset);
+		hva = hpa2hva(addr);
+
+		switch (bytes) {
+		case 1U:
+			mmio_write8((uint8_t)val, hva);
+			break;
+		case 2U:
+			mmio_write16((uint16_t)val, hva);
+			break;
+		default:
+			mmio_write32(val, hva);
+			break;
+		}
+	} else {
+		pr_err("the offset should be aligned with 2/4 byte\n");
```

Comment:
Same question.  Will this message make any sense without more context?

Reply:
@dbkinder  thank you, after re-think this fix, seems need to double check and validate these inputs in the caller of these functions, if so, we don't need to break the @pre of them.

so, pls hold on this PR for now, sorry for this confusing.